libs/cardano-ledger-conformance/cardano-ledger-conformance.cabal

```
@@ -33,16 +33,37 @@ library
         bytestring,
         cardano-strict-containers,
         microlens,
+        mtl,
+        data-default-class,
         cardano-ledger-binary,
         cardano-ledger-core,
         cardano-ledger-allegra,
-        cardano-ledger-shelley,
+        cardano-ledger-mary,
+        cardano-ledger-shelley:{cardano-ledger-shelley, testlib},
         cardano-ledger-alonzo,
-        cardano-ledger-conway,
+        cardano-ledger-conway:{cardano-ledger-conway, testlib},
         cardano-ledger-executable-spec,
         cardano-crypto-class,
+        cardano-ledger-core:testlib,
+        cardano-ledger-test,
         containers,
+        small-steps,
         text
 
     if flag(asserts)
         ghc-options: -fno-ignore-asserts
+
+test-suite tests
+    type:             exitcode-stdio-1.0
+    main-is:          Main.hs
+    hs-source-dirs:   test
+    default-language: Haskell2010
+    ghc-options:
+        -Wall -Wcompat -Wincomplete-record-updates
+        -Wincomplete-uni-patterns -Wredundant-constraints -Wpartial-fields
+        -Wunused-packages -threaded -rtsopts -with-rtsopts=-N
+
+    build-depends:
+        base >=4.14 && <4.19,
```

Upper bound on `base` is dumb. Have no idea why people add it
```suggestion
        base >=4.14 && <5,
```